implementations/rust/ockam/ockam_abac/src/policy.rs

```diff
@@ -82,10 +126,134 @@ impl IncomingAccessControl for PolicyAccessControl {
         AbacAccessControl::new(
             self.identities_attributes.clone(),
             self.authority.clone(),
-            policy,
+            policy_expression,
             self.environment.clone(),
         )
         .is_authorized(msg)
         .await
     }
 }
+
+pub struct Policies {
+    resources_repository: Arc<dyn ResourcePoliciesRepository>,
+    resource_types_repository: Arc<dyn ResourceTypePoliciesRepository>,
+}
+
+impl Policies {
+    pub fn new(
+        resources_repository: Arc<dyn ResourcePoliciesRepository>,
+        resource_types_repository: Arc<dyn ResourceTypePoliciesRepository>,
+    ) -> Self {
+        Self {
+            resources_repository,
+            resource_types_repository,
+        }
+    }
+
+    async fn get_expression_for_resource_type_and_name(
+        &self,
+        resource: &Resource,
+        action: &Action,
+    ) -> Result<Expr> {
+        let mut expression = if let Some(policy) = self
+            .resources_repository
+            .get_policy(&resource.resource_name, action)
+            .await?
+        {
+            policy.into_expression()
+        } else {
+            return Err(Error::new(
```

Comment:
I'm not sure I understand this one. So if I have a resource type policy defined, but don't have a policy for a specific resource - this function errors? I would expect we just use resource type policy in this case

Reply:
At this point I was expecting that we would have at least the `check credential` policy that we use as a fallback. I'm worried we could be hiding a misconfiguration bug of the resources if we fallback to the resource type policy.

Reply:
I've also changed this and it now behaves as you said.